internal/transport/http2_client.go

```diff
@@ -1279,59 +1296,112 @@ func (t *http2Client) reader() {
 	}
 }
 
-// keepalive running in a separate goroutune makes sure the connection is alive by sending pings.
+// activityMonitory reads from the activity channel (which is written to, when
+// there is a read), and updates the lastRead.time atomic.
+func (t *http2Client) activityMonitor() {
+	for {
+		select {
+		case <-t.lr.ch:
+			atomic.StoreInt64(&t.lr.time, time.Now().Unix())
```

Comment:
As we already have an int64, should we store last read time with a higher precision?

Reply:
I decided to go with `second` precision since that's what the proposal asks for, but we could as well store `time.Now().UnixNano()` if that is preferable.